BepInEx.Core/Paths.cs

```diff
@@ -106,9 +106,20 @@ public static void SetExecutablePath(string executablePath,
                            ? Utility.ParentDirectory(executablePath, 4)
                            : Path.GetDirectoryName(executablePath);
 
-        GameDataPath = managedPath != null && gameDataRelativeToManaged
-                           ? Path.GetDirectoryName(managedPath)
-                           : Path.Combine(GameRootPath, $"{ProcessName}_Data");
+        if (managedPath != null && gameDataRelativeToManaged)
+        {
+            GameDataPath = Path.GetDirectoryName(managedPath);
+        }
+        else
+        {
+            GameDataPath = Path.Combine(GameRootPath, $"{ProcessName}_Data");
```

Comment:
```suggestion
            // According to some experiments, Unity checks whether globalgamemanagers/data.unity3d exists in the data folder before picking it.
            // 'ProcessName_Data' folder is checked first, then if that fails 'Data' folder is checked. If neither is valid, the player crashes.
            // A simple Directory.Exists check is accurate enough while being less likely to break in case these conditions change.
            GameDataPath = Path.Combine(GameRootPath, $"{ProcessName}_Data");
```